### ISSUE TITLE
SwiftLint Incompatible with SwiftFormat

### ISSUE THREAD

**Issue by Mordil** [realm/SwiftLint#4952]
### New Issue Checklist

- [X] Updated SwiftLint to the latest version
- [X] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

When trying to install the latest version of SwiftLint and [apple/SwiftFormat](https://github.com/apple/swift-format), SwiftPM complains about incompatible versions for SwiftSyntax, so package resolution fails.

**Why is SwiftLint using a development snapshot for an upcoming version of Swift to power the tool?**

It's completely understandable for development, but full releases should be relying on the latest stable release, which as of writing is `508.0.0`

##### Complete output from SwiftPM

```bash
Failed to resolve dependencies Dependencies could not be resolved because 'socrates' depends on 'swift-format' 508.0.0 and 'acropolis' depends on 'swiftlint' 0.51.0..<0.52.0.
'swiftlint' is incompatible with 'swift-format' because 'swift-format' 508.0.0 depends on 'swift-syntax' 508.0.0..<509.0.0.
'swiftlint' >= 0.51.0 practically depends on 'swift-syntax' 0.50900.0-swift-DEVELOPMENT-SNAPSHOT-2023-02-06-a because 'swiftlint' 0.51.0 depends on 'swift-syntax' 0.50900.0-swift-DEVELOPMENT-SNAPSHOT-2023-02-06-a and no versions of 'swiftlint' match the requirement 0.51.1..<0.52.0.
```

### Environment

* SwiftLint `0.51.0`
* Swift Package Manager, build plugin
* Xcode 14.3 (14E222b)
* SwiftFormat `508.0.0`

**Comment by SimplyDanny** [comment]
You presumably mean [apple/swift-format](https://github.com/apple/swift-format). Right?

**Comment by Mordil** [comment]
> You presumably mean [apple/swift-format](https://github.com/apple/swift-format). Right?

Yes, apologies, I should've added that clarification

**Comment by AndrewSB** [comment]
hey! similar failure for when trying to install alongside https://github.com/uber/mockolo. they depend on `.exact(508.0.0)`

**Comment by Brumanuel** [comment]
Similar failure for when trying to install https://github.com/Brightify/Cuckoo 

`Failed to resolve dependencies Dependencies could not be resolved because root depends on 'cuckoo' 2.0.9 and root depends on 'swiftlint' 0.55.1..<1.0.0.
'swiftlint' is incompatible with 'cuckoo' because 'cuckoo' 2.0.9 depends on 'swift-syntax' 509.0.0.
'swiftlint' >= 0.55.1 practically depends on 'swift-syntax' 510.0.2..<511.0.0 because 'swiftlint' 0.55.1 depends on 'swift-syntax' 510.0.2..<511.0.0 and no versions of 'swiftlint' match the requirement 0.55.2..<1.0.0.
`

**Comment by SimplyDanny** [comment]
> Why is SwiftLint using a development snapshot for an upcoming version of Swift to power the tool?

I could elaborate on that and list many points about why using a development snapshot and a fixed version in the first place is actually a good thing. I'll go with a rather shortest answer, though, which is: Use the plugin from the [dedicated repository](https://github.com/SimplyDanny/SwiftLintPlugins) to avoid these issues and benefit from the significant performance improvements only possible by relying on a development snapshot of SwiftSyntax. 😉
